autoscale_cloudcafe/autoscale/behaviors.py

```diff
@@ -538,3 +578,90 @@ def get_webhooks_properties(self, webhook_list):
                 pass
             webhook['count'] = len(webhook_list)
             return webhook
+
+    def wait_for_expected_number_of_active_servers(self, group_id, expected_servers,
+                                                   interval_time=None, timeout=None,
+                                                   api="Autoscale", asserter=None):
+        """
+        Wait for the expected_servers to arrive in either Autoscale or RackConnectV3 API.
+
+        :summary: verify the desired capacity in group state is equal to expected servers
+         and waits for the specified number of servers to be active on a group
+        :param group_id: Group id (AutoScale API), or pool ID (RackConnect API)
+        :param expected_servers: Total active servers expected on the group
+        :param interval_time: Time to wait during polling group state
+        :param timeout: Time to wait before exiting this function
+        :param api: Either "Autoscale" or "RackConnect".  Defaults to "Autoscale"
+        :param asserter: Object responsible for enforcing invariants through assertions.
+                         If none provided, a default do-nothing asserter will be assumed.
+                         You won't be able to tell if things pass or fail, though.
+                         It's best if you pass in your own asserter.
+        :return: returns the list of active servers in the group
+        """
+
+        interval_time = interval_time or int(
+            self.autoscale_config.interval_time)
+        timeout = timeout or int(self.autoscale_config.timeout)
+        start_time = time.time()
+        end_time = start_time + timeout
+
+        # If we didn't receive an asserter, let's assume a do-nothing asserter.
+        if asserter is None:
+            asserter = DefaultAsserter()
+
+        # If we're checking up on Autoscale, we need to make sure that our
+        # group is even attempting to build the right number of servers.  If
+        # we're not, there's no point in continuing; we'll never attain the
+        # desired capacity anyway.
+        #
+        # Unfortunately, there's no way to check this on RackConnect's API,
+        # since it has no mechanism to report back its current "plans."
+        if api == 'Autoscale':
```

Comment:
This `if` block broke the tests. It is only checking if desired matches and returns instead of ensuring active list is reached. Also, I don't understand its purpose. Is it meant to be an optimization by checking desired first? If so, that is already done line 641 below.